src/native-stack/types.tsx

```diff
@@ -112,6 +112,15 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   disableBackButtonMenu?: boolean;
+  /**
+   * How the back button behaves by default (when not customized). Available on iOS>=14.
+   * The following values are currently supported (they correspond to https://developer.apple.com/documentation/uikit/uinavigationitembackbuttondisplaymode?language=objc):
+   * - "default" – shows given back button title/previous controller title, system default or just icon based on available space
+   * - "generic" – shows given system default or just icon based on available space
+   * - "minimal" – shows just an icon
+   * @platform ios
+   */
+  backButtonDisplayMode?: ScreenStackHeaderConfigProps['backButtonDisplayMode'];
```

Comment:
We need to also think through how this property interacts with `backTitleVisible` prop. 
What happens if we set `backTitleVisible: false` and set this to default or other options? And the other way around.

Reply:
At least we should document these interactions between any related props here.

Reply:
I don't think that the logic changes significantly in this case, pls look at:
<img width="1089" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/11800297/645f815e-c819-4a0b-a834-2acc9e6aa82a">
This is how it worked before, we set prevItem.backButtonTitle for back button menu when backTitleVisible was false.

Reply:
Ok, I actually revert bunch of this changes. We can't use backButtonTitle to set custom label - more details in PR summary.